feat: enable collection search extension

stephenkilbourn opened the pull request:
### Issue

#495 

### What?
- Adds the Collection Search functionality to STAC API
- Slight refactor to api configuration to better align with example app in stac-fastapi.pgstac

anayeaye commented:
@hrodmn Had just discovered the separate image for pypgstac...might reach out when trying to add https://github.com/stac-utils/pgstac/pkgs/container/pgstac-pypgstac this afternoon. Looking at https://github.com/stac-utils/pgstac/blob/main/docker-compose.yml

hrodmn commented:
@anayeaye I made some progress too, there is a failing mosaic test but I think the basic mechanics are working. I think we would just need to make sure CI can use the new pypgstac container instead of running pypgstac in the veda.db container.

<details><summary>my diff so far</summary>
<p>

```diff
diff --git a/.github/workflows/data/noaa-emergency-response.json b/.github/workflows/data/noaa-emergency-response.json
index 3d426cc..989bf7a 100644
--- a/.github/workflows/data/noaa-emergency-response.json
+++ b/.github/workflows/data/noaa-emergency-response.json
@@ -1,28 +1,17 @@
 {
-    "id": "noaa-emergency-response",
-    "title": "NOAA Emergency Response Imagery",
-    "description": "NOAA Emergency Response Imagery hosted on AWS Public Dataset.",
-    "stac_version": "1.0.0",
-    "license": "public-domain",
-    "links": [],
-    "extent": {
-        "spatial": {
-            "bbox": [
-                [
-                    -180,
-                    -90,
-                    180,
-                    90
-                ]
-            ]
-        },
-        "temporal": {
-            "interval": [
-                [
-                    "2005-01-01T00:00:00Z",
-                    "null"
-                ]
-            ]
-        }
+  "id": "noaa-emergency-response",
+  "title": "NOAA Emergency Response Imagery",
+  "description": "NOAA Emergency Response Imagery hosted on AWS Public Dataset.",
+  "stac_version": "1.0.0",
+  "license": "public-domain",
+  "links": [],
+  "extent": {
+    "spatial": {
+      "bbox": [[-180, -90, 180, 90]]
+    },
+    "temporal": {
+      "interval": [["2005-01-01T00:00:00Z", null]]
     }
-}
\ No newline at end of file
+  }
+}
+
diff --git a/docker-compose.yml b/docker-compose.yml
index 864b58f..5c8b52f 100644
--- a/docker-compose.yml
+++ b/docker-compose.yml
@@ -106,7 +106,7 @@ services:
       - PGPASSWORD=password
       - PGDATABASE=postgis
     ports:
-      - "5432:5432"
+      - "5439:5432"
     command: postgres -N 500
     volumes:
       - ./scripts:/tmp/scripts
@@ -114,6 +114,25 @@ services:
     # broken in github actions (definitely when run in act, possibly in tests involving ingest-api)? re-enable to persist local database
     #  - ./.pgdata:/var/lib/postgresql/data
 
+  pypgstac:
+    container_name: veda.loadtestdata
+    platform: linux/amd64
+    image: ghcr.io/stac-utils/pgstac-pypgstac:v0.9.6
+    environment:
+      - PGUSER=username
+      - PGPASSWORD=password
+      - PGDATABASE=postgis
+      - PGHOST=database
+      - PGPORT=5432
+    depends_on:
+      - database
+    command: bash -c "sleep 10 && /tmp/scripts/bin/load-data.sh"
+    volumes:
+      - ./scripts:/tmp/scripts
+      - ./.github/workflows/data:/tmp/data
+
   ingestor:
     container_name: veda.ingestor
     platform: linux/amd64
@@ -128,7 +147,7 @@ services:
       - PGPASSWORD=password
       - PGDATABASE=postgis
       - DYNAMODB_ENDPOINT=http://localhost:8085
-      - VEDA_DB_PGSTAC_VERSION=0.7.10
+      - VEDA_DB_PGSTAC_VERSION=0.9.6
       - AWS_REGION=us-west-2
       - AWS_DEFAULT_REGION=us-west-2
       - DYNAMODB_TABLE=veda
diff --git a/scripts/bin/load-data.sh b/scripts/bin/load-data.sh
index 8caa4aa..28842e9 100755
--- a/scripts/bin/load-data.sh
+++ b/scripts/bin/load-data.sh
@@ -3,17 +3,17 @@
 set -e
 
 # Wait for the database to start
-until pypgstac pgready --dsn postgresql://username:password@0.0.0.0:5432/postgis; do
+until pypgstac pgready; do
   echo "Waiting for database to start..."
   sleep 1
 done
 
 # Load collections
 echo "Loading collections..."
-pypgstac load collections /tmp/data/noaa-emergency-response.json --dsn postgresql://username:password@0.0.0.0:5432/postgis --method upsert
+pypgstac load collections /tmp/data/noaa-emergency-response.json --method upsert
 
 # Load items
 echo "Loading items..."
-pypgstac load items /tmp/data/noaa-eri-nashville2020.json --dsn postgresql://username:password@0.0.0.0:5432/postgis --method upsert
+pypgstac load items /tmp/data/noaa-eri-nashville2020.json --method upsert
 
 echo "Data loaded successfully!"
diff --git a/scripts/run-local-tests.sh b/scripts/run-local-tests.sh
index eccd184..2d94a32 100755
--- a/scripts/run-local-tests.sh
+++ b/scripts/run-local-tests.sh
@@ -5,7 +5,9 @@ set -e
 pre-commit run --all-files
 
 # Bring up stack for testing; ingestor not required
-docker compose up -d stac raster database dynamodb
+docker compose up -d --wait stac raster database dynamodb pypgstac 
+
+echo "waiting for applications to start" && sleep 10
 
 # cleanup, logging in case of failure
 cleanup() {
@@ -25,9 +27,6 @@ cleanup() {
 }
 trap cleanup EXIT
 
-# Load data for tests
-docker exec veda.db /tmp/scripts/bin/load-data.sh
-
 # Run tests
 python -m pytest .github/workflows/tests/ -vv -s

```

</p>
</details>

anayeaye commented:
UPDATE: I pushed @hrodmn's docker-compose and test changes which get us running tests up until the transactions which I still haven't figured out. Those tests are failing to connect to the db container and local runs are failing on Failed: async def functions are not natively supported. The transactions tests apparently have been running up until now though

anayeaye commented:
🙈 the docker compose database port is 5439

stephenkilbourn commented:
@anayeaye @hrodmn - i refactored a little to use a `collection_in_db fixture` to make sure we create a collection for each test, and also added some tests cases around the search endpoint.

anayeaye commented on scripts/load-data-container.sh:
we load the container in docker compose now

anayeaye commented on scripts/run-local-tests.sh:
```diff
@@ -26,7 +26,7 @@ cleanup() {
 
 # Load data for tests
-docker exec veda.db /tmp/scripts/bin/load-data.sh
+docker exec veda.loadtestdata /tmp/scripts/bin/load-data.sh
```
I guess we don't need this anymore but I can't decide if it is still worth having here as pseudo documentation

anayeaye commented:
> @stephenkilbourn @hrodmn
> 
> I pushed some changes to clean out some commented code that I had pushed. I still can't successfully run run-local-tests.sh and I am fairly sure that it something local but can you confirm that this file runs for you?

I did a docker prune and tests are now running for me. Please disregard.
